lib/engine.rb

```diff
@@ -1,64 +1,130 @@
 # frozen_string_literal: true
 
-# Class for game state
 class Game
   attr_reader :state
 
   def initialize
     @state = true
+    @counter = 0
+  end
+
+  def increase_counter
+    @counter += 1
   end
 
-  # state to finish games
   def finish_game
     @state = false
   end
+
+  def line_checker(board, winner)
+    return unless @counter >= 4
+
+    vertical = vertical(board)
+
+    horizontal = horizontal(board)
+
+    diagonal = diagonal(board)
+
+    if vertical || horizontal || diagonal
+      finish_game
+      puts "Player #{winner} won!"
+    elsif @counter == 8
+      puts 'Draw!'
+    end
+  end
+
+  def vertical(board)
+    board[0][0].state == board[1][0].state && board[1][0].state == board[2][0].state ||
+      board[0][1].state == board[1][1].state && board[1][1].state == board[2][1].state ||
+      board[0][2].state == board[1][2].state && board[1][2].state == board[2][2].state
+  end
+
+  def horizontal(board)
+    board[0][0].state == board[0][1].state && board[0][1].state == board[0][2].state ||
+      board[1][0].state == board[1][1].state && board[1][1].state == board[1][2].state ||
+      board[2][0].state == board[2][1].state && board[2][1].state == board[2][2].state
+  end
+
+  def diagonal(board)
+    board[0][0].state == board[1][1].state && board[1][1].state == board[2][2].state ||
+      board[0][2].state == board[1][1].state && board[1][1].state == board[2][0].state
+  end
 end
 
 class Cell
-  attr_reader :state
-
   def initialize
     @state = nil
   end
 
-  # Method to mark X on board
+  attr_reader :state
+
+  attr_writer :state
+
   def make_cross
     @state = 'X'
   end
 
-  # Method to mark O on board
   def make_o
     @state = 'O'
   end
 end
 
-# Tic Tack Toe grid
 class Board
```

Comment:
[OPTIONAL]
Maybe try putting this class in a separate file as it is better separation of code.
If you really want to have it in this engine file you can always require it